smdebug/profiler/python_profiler.py

```diff
@@ -225,6 +227,8 @@ class PyinstrumentPythonProfiler(PythonProfiler):
     def _enable_profiler(self):
         """Enable the pyinstrument profiler.
         """
+        if sys.getprofile() is not None:
```

Comment:
Why is this change required to support mirrored strategy?
Add a comment if this is explicitly required for distributed training.

Reply:
See PR description.

> Fix bug in Pyinstrument profiling to support distributed training

Reply:
Add comments explaining the underlying logic.

Reply:
Decided to dive deeper and realized that it doesn't make sense for some python profiling functionality (profiling during the step) to be distributed across the workers and for some python profiling functionality (pre-step-zero and post-hook-close) to be called before and after training. All python profiling calls need to be distributed across all of the workers. Once I made that change, I no longer saw this edge case occurring and removed this condition.

TL;DR wrote an incorrect use case earlier, this edge case is no longer valid.

Reply:
Remove this code snippet then